File: ios-swiftUI/Tday/Feature/Guide/GuideHelpLink.swift
import SwiftUI

/// Opens the How-To guide pre-scrolled to a topic id. Set once at the app root
/// (AppRootView); the nil default hides every [GuideHelpLink] in previews or
/// surfaces without the main navigation stack.
private struct OpenGuideTopicKey: EnvironmentKey {
    static let defaultValue: ((String) -> Void)? = nil
}

extension EnvironmentValues {
    var openGuideTopic: ((String) -> Void)? {
        get { self[OpenGuideTopicKey.self] }
        set { self[OpenGuideTopicKey.self] = newValue }
    }
}

/// Topic ids mirrored from the shared `GuideTopicIds` (content is
/// contract-tested against the bundled artifact, which carries these ids).
enum GuideTopicId {
    static let nlpDateSyntax = "nlp-date-syntax"
    static let recurrencePresets = "recurrence-presets"
}

/// A quiet contextual "?" that deep-links from a feature surface into its
/// guide topic — the iOS counterpart of the web/Android `GuideHelpLink`.
/// Uses a tap gesture (not a Button) so it stays tappable when embedded
/// inside a larger Button row, and dismisses the presenting sheet before
/// pushing the guide onto the navigation stack.
struct GuideHelpLink: View {
    let topicId: String

    @Environment(\.openGuideTopic) private var openGuideTopic
    @Environment(\.dismiss) private var dismiss
    @Environment(\.tdayColors) private var colors

    var body: some View {
        if let openGuideTopic {
            Image("LucideCircleHelp")
                .renderingMode(.template)
                .resizable()
                .scaledToFit()
                .frame(width: 18, height: 18)
                .foregroundStyle(colors.onSurfaceVariant.opacity(0.85))
                .padding(6)
                .contentShape(Rectangle())
                .onTapGesture {
                    dismiss()
                    openGuideTopic(topicId)
                }
                .accessibilityLabel(Text(L("How-To & Tips")))
                .accessibilityAddTraits(.isButton)
        }
    }
}

File: android-compose/app/src/main/java/com/ohmz/tday/compose/feature/guide/GuideHelpLink.kt
package com.ohmz.tday.compose.feature.guide

import androidx.compose.foundation.layout.size
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.staticCompositionLocalOf
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.res.vectorResource
import androidx.compose.ui.unit.dp
import com.ohmz.tday.compose.R

/**
 * Opens the How-To guide pre-scrolled to a topic id (from the shared
 * `GuideTopicIds`). Provided once in TdayApp around the NavHost; null wherever
 * there is no navigation host (e.g. the widget quick-add activity), which
 * simply hides every [GuideHelpLink].
 */
val LocalOpenGuideTopic = staticCompositionLocalOf<((String) -> Unit)?> { null }

/**
 * A quiet contextual "?" that deep-links from a feature surface into its guide
 * topic — the Android counterpart of the web `GuideHelpLink`. Renders nothing
 * when no [LocalOpenGuideTopic] provider is in scope.
 */
@Composable
fun GuideHelpLink(topicId: String, modifier: Modifier = Modifier) {
    val openGuideTopic = LocalOpenGuideTopic.current ?: return
    IconButton(
        onClick = { openGuideTopic(topicId) },
        modifier = modifier.size(32.dp),
    ) {
        Icon(
            imageVector = ImageVector.vectorResource(R.drawable.ic_lucide_circle_help),
            contentDescription = stringResource(R.string.settings_help_guide),
            tint = MaterialTheme.colorScheme.onSurfaceVariant,
            modifier = Modifier.size(18.dp),
        )
    }
}
